File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dummy = new ListNode();
        ListNode current = dummy;

        while(list1 != null && list2 != null) {
            int v1 = list1.val;
            int v2 = list2.val;

            if(v1 >= v2) {
                ListNode new_node = new ListNode(v2);
                current.next = new_node;
                current = current.next;
                list2 = list2.next;
            }
            if (v1 < v2) {
                ListNode new_node = new ListNode(v1);
                current.next = new_node;
                current = current.next;
                list1 = list1.next;
            }
        }
            if(list1 != null) {
                current.next = list1;
            }
            if(list2 != null) {
                current.next = list2;
            }
        return dummy.next;
    }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode()
        cur = dummy
        
        while l1 and l2:
            if l1.val <= l2.val:
                cur.next = ListNode(l1.val)
                cur = cur.next
                l1 = l1.next
            else:
                cur.next = ListNode(l2.val)
                cur = cur.next
                l2 = l2.next
        
        if l1:
            cur.next = l1
        if l2:
            cur.next = l2
        return dummy.next